### PR TITLE
Upgrade got to version 11 beta

### PR DESCRIPTION
**Pull request by larixer**
**What's the problem this PR addresses?**

Berry issues many network requests and DNS lookups. In case if underlying OS do not have DNS cache enabled this might lead in some time waste during resolution and fetch steps.

**How did you fix it?**

Upgraded got to version 11, which enables caching DNS lookups by default. It also lets us remove hooks to handle authorization header leakage, it doesn't happen anymore with got 11.

### CODE REVIEW

**Comment by larixer** [comment]
Mac OS X integration tests consistently fail with timeouts... But looks like it is not the result of dnsCache, trying to figure out the reason

**Comment by larixer** [comment]
Found out the reason for integration tests failure under OS X and submitted issue for `cacheable-lookup`:
https://github.com/szmarczak/cacheable-lookup/issues/13

**Comment by szmarczak** [comment]
The cause for this was the `cacheable-lookup@latest` tag pointing to an outdated version. I haven't altered with it in any way, so that's weird. Fixed.

**Comment by merceyz** [comment]
First version of Got to use `cacheable-lookup@^4` is `got@11.0.0-beta.1`

**Comment by szmarczak** [comment]
> If there are no changes that would break Got we can force an update using packageExtensions

It won't work as `got@10.7.0` does `new CacheableLookup({cacheAdapter: options.dnsCache})` but `cacheAdapter` has been renamed to `cache`, so instead `{dnsCache}` you need to pass

```js
{
	dnsCache: new CacheableLookup({
		cache: dnsCache
	})
}
```

**Comment by szmarczak** [comment]
Also I would love to hear some feedback on the beta. If any of the tests fail with the new beta, post an issue straight away :)

File: packages/yarnpkg-core/sources/httpUtils.ts [hunk 1]
```diff
@@ -86,16 +83,16 @@ export async function request(target: string, body: Body, {configuration, header
...
+      }) as BeforeRequestHook,
     ],
     beforeRedirect: [
-      (options: NormalizedOptions) => {
+      ((options: NormalizedOptions) => {
         if (options.headers && options.headers.authorization && options.url.hostname !== hostname) {
           delete options.headers.authorization;
         }
-      },
+      }) as unknown as BeforeRedirectHook,
```
Comment (szmarczak) [inline packages/yarnpkg-core/sources/httpUtils.ts:95]: The `authorization` header leak was fixed in the beta, you can remove this now :)

Comment (larixer) [reply packages/yarnpkg-core/sources/httpUtils.ts:95]: Okay, will try now, thanks!
One minor note, I have noticed that `beforeRedirectHook` uses the different type of `NormalizedOptions` than `NormalizedOptions` exported from top-level got `index.d.ts`, thats why I had to cast to `as unknown as BeforeRedirectHook` to shut up `tsc`.
image

**Comment by szmarczak** [comment]
Got 11 is now stable! Also released `11.0.1`, which fixes two small regressions :)

**Pull request merged**